crates/core/src/types/field_element.rs

```diff
@@ -1,5 +1,16 @@
 use starknet::core::types::FieldElement;
 
+pub trait FieldElementVecExt {
+    fn extend_with_len(&mut self, other: &[FieldElement]);
+}
+
+impl FieldElementVecExt for Vec<FieldElement> {
+    fn extend_with_len(&mut self, other: &[FieldElement]) {
+        self.push(FieldElement::from(other.len() as u64));
+        self.extend(other.clone());
+    }
+}
+
 pub trait IntoFieldElementVec {
```

Comment:
We should renamed it `ToFieldElementVec`

Reply:
Updated!